armi/reactor/reactors.py

```diff
@@ -2316,3 +2326,12 @@ def processLoading(self, cs, dbLoad: bool = False):
         self.p.maxAssemNum = self.getMaxParam("assemNum")
 
         getPluginManagerOrFail().hook.onProcessCoreLoading(core=self, cs=cs)
+
+    def updateBlockBOLHeights(self):
+        """post thermal expansion, update block BOL heights"""
+        self.p.referenceBlockAxialMesh = self.findAllAxialMeshPoints(applySubMesh=False)
+        self.p.axialMesh = self.findAllAxialMeshPoints()
+        for a in self.getAssemblies():
```

Comment:
this should be `for a in assemsToAxiallyExpand`

Reply:
Addressed https://github.com/terrapower/armi/pull/823/commits/5ad8f492ef706848ec14c6e7d8c3df676113bc3c